core/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1460,11 +1460,16 @@ public static double pow(final double a, final double b) {
      * Returns the distance between two GeoPointData locations, in meters, given objects to unpack.
      * Ignores altitude and accuracy.
      * Note that the arguments can be strings.
+     * Returns -1 if one of the arguments is null or the empty string.
      */
     public static Double distance(Object from, Object to) {
         String unpackedFrom = (String) unpack(from);
         String unpackedTo = (String) unpack(to);
 
+        if (unpackedFrom == "" || unpackedFrom == null || unpackedTo == "" || unpackedTo == null) {

```

Comment:
`someString == ""` does a reference level equality check in Java, which isn't what you want here. 
Should be `someString == null || "".equals(someString)`